weld-rendering/app/src/main/java/styler/weld/rendering/viewholder/WidgetFactory.kt

```diff
@@ -1,31 +1,60 @@
 package styler.weld.rendering.viewholder
 
+import android.view.LayoutInflater
 import android.view.ViewGroup
+import styler.weld.rendering.R
 
 class WidgetFactory {
 
     var factories: MutableList<FactoryMethod> = mutableListOf();
 
-    fun register(type: String, factory: (parent: ViewGroup) -> BaseViewHolder) {
-        factories.add(FactoryMethod(type, factory));
+    fun register(type: String) {
+        factories.add(FactoryMethod(type))
     }
 
-    fun createViewHolder(parent: ViewGroup, viewType: Int): BaseViewHolder {
+    fun createViewHolder(viewType: Int, parent: ViewGroup): BaseViewHolder {
+        if (viewType == INVALID) {
+            // TODO: INVALID case should be refactored later
+            val view = LayoutInflater.from(parent.context)
+                .inflate(R.layout.empty_view_row_item, parent, false)
+            return EmptyViewHolder(view)
+        }
         return factories[viewType].createViewHolder(parent)
```

Comment:
I encountered an index exception where the list index is passed as `-1` (because the `INVALID` value is `-1`). Made a quick fix for this here.

Reply:
Good call 👍

Reply:
Refactored invalid type in here https://github.com/ludugz/weld-rendering/pull/2/commits/e3f7636caa3fd24e4ea7c60b5149e57ff0e3082e